File: Code/0217_Contains_Duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashMap<Integer, Integer> hmNums = new HashMap<Integer, Integer>();
        
        for(int i = 0; i < nums.length; i++)
        {
            //System.out.println(nums[i] + " key:: " + hmNums.get(nums[i]));
            if(hmNums.get(nums[i]) == null)
                hmNums.put(nums[i], nums[i]);
            else
                return true;
        }
        
        return false;
    }
}

File: Code/0217_Contains Duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        map<int, int> nexist;

        for(auto i = 0; i < nums.size(); i++)
        {
            if(nexist.find(nums[i]) != nexist.end())
                return true;
            else
                nexist[nums[i]] = nums[i];
        }

        return false;
    }
};